stack.hpp

```diff
@@ -16,17 +16,19 @@ class Stack {
     }
 
     void push(const T& element) {
-        if (position_ > capacity_) throw std::out_of_range("Not enough capacity");
-        storage_[position_++] = element;
+        if (position_ >= capacity_) throw std::out_of_range("Not enough capacity");
+        *(storage_ + ++position_) = element;
```

Comment:
No, when pre-incrementing the position at insertion you will write outside the storage_ memory and you will get undefined behaviour. Consider post-increment for the push() method and pre-increment for the pop() method.